src/Server.cpp

```diff
@@ -1,41 +1,85 @@
 #include <thread>
 #include <iostream>
+#include <algorithm>
+#include <signal.h>
 
 #include "server.hpp"
+#include "exceptions.hpp"
+
+static Server *_instance;
```

Comment:
this could ez pizy be a shared ptr

Reply:
But then the constructor has to be public, no?

Reply:
I read about the solution. A bit ugly but sure...